src/box/xrow_update_field.c

```diff
@@ -237,14 +237,25 @@ xrow_update_mp_read_scalar(const char **expr,
 		ret->type = XUPDATE_TYPE_FLOAT;
 		ret->flt = mp_decode_float(expr);
 		break;
-	case MP_STR: {
-		ret->type = XUPDATE_TYPE_STR;
-		struct xrow_update_string *str = &ret->str;
-		str->array[0].data = mp_decode_str(expr, &str->array[0].length);
+	case MP_STR:
+	case MP_BIN:
+	{
+		ret->type = XUPDATE_TYPE_STRBIN;
+		struct xrow_update_strbin *str = &ret->strbin;
+		if (mp_typeof(**expr) == MP_STR) {
+			str->array[0].data =
+				mp_decode_str(expr, &str->array[0].length);
+			str->store_type = XUPDATE_STORE_TYPE_STR;
+		} else {
+			assert(mp_typeof(**expr) == MP_BIN);
+			str->array[0].data =
+				mp_decode_bin(expr, &str->array[0].length);
+			str->store_type = XUPDATE_STORE_TYPE_BIN;
+		}
 		memset(&str->array[1], 0, sizeof(str->array[1]) * 2);
-		ret->str.type = XROW_UPDATE_STRING_SINGLE;
+		str->type = XROW_UPDATE_STRING_SINGLE;
```

Comment:
I guess it is worth rename `xrow_update_string_type` also.

Reply:
ok, done.

Reply:
Let's then rename `enum xrow_update_strbin_type` values too.

I don't know, if we consistently update all names then we need to rename so many constants, struct, comments and variables. `xrow_update_string_chunk`, `xrow_update_string_rope`, change string to `string (or varbinary)` in comments, rename `str` vars to `strbin` etc etc.

Anyway I guess either we should do it or keep string name everywhere. What do you think?

Reply:
Actually I think that rename of `xrow_update_string_type` was excess. Let's rename it back?
The reason is that we actually load MP_BIN value to string, aka sequence of chars. From that point we work with thing as with string. That thing can consist of one or several sustrings etc. So we should name struct type and the corresponding member of scalar as *strbin to be sure we will not forget to store is as MP_BIN in the end. The all other names may remain string-like. As for local 'str' names, that does not matter much. It could be 's' or whatereber.

Reply:
Discusses offline. Decided not to use 'strbin' at all.